phase/download_binaries.go

```diff
@@ -89,12 +90,21 @@ type binary struct {
 }
 
 func (b *binary) download() error {
-	path, err := cache.GetOrCreate(b.downloadTo, "k0s", b.os, b.arch, "k0s-"+b.version+b.ext())
```

Comment:
this baffles me. why did it have a function name `b.downloadTo` there? how did it ever work? when was the download function actually called?  WEIRD!

Reply:
ahh, the cache thing accepts a function as the first parameter. very super weird.